common/src/test/java/com/revenuecat/purchases/common/BackendTest.kt

```diff
@@ -1451,6 +1492,202 @@ class BackendTest {
         assertThat(headersSlot.captured["marketplace"]).isEqualTo("DE")
     }
 
+    // region postDiagnostics
+
+    @Test
+    fun `postDiagnostics makes call with correct parameters`() {
+        val diagnosticsList = listOf(JSONObject("{\"test-key\":\"test-value\"}"))
+        backend.postDiagnostics(diagnosticsList, {}, { _, _ -> })
+        verify(exactly = 1) {
+            mockClient.performRequest(
+                baseURL = mockDiagnosticsBaseURL,
+                path = "/telemetry",
+                body = mapOf("entries" to JSONArray(diagnosticsList)),
+                requestHeaders = mapOf("Authorization" to "Bearer TEST_API_KEY")
+            )
+        }
+    }
+
+    @Test
+    fun `postDiagnostics only executes once same request if one in progress`() {
+        val diagnosticsList = listOf(JSONObject("{\"test-key\":\"test-value\"}"))
+        mockResponse(
+            path = "/telemetry",
+            body = null,
+            responseCode = 200,
+            clientException = null,
+            resultBody = null,
+            delayed = true,
+            baseURL = mockDiagnosticsBaseURL
+        )
+        val lock = CountDownLatch(3)
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        asyncBackend.postDiagnostics(diagnosticsList, { lock.countDown() }, { _, _ -> fail("expected success") })
+        lock.await(2000, TimeUnit.MILLISECONDS)
```

Comment:
Does this block the test for 2 seconds?

Reply:
No, this is just the timeout we will hold to wait for the lock to be released. When we add the `delayed` parameter in the `mockResponse` method, that waits 200ms, so this should wait ~200ms.

Reply:
What's kind of a lot, 5 of those is a whole second. It would work with a delay of 20ms too, right?

Reply:
Actually it's only 200ms, since it would basically execute all 3 at the same time and just wait for the first one. We could potentially lower it I guess, that's the default that was setup for the other tests, but could try to lower it and see if it doesn't affect the tests.

Reply:
Reduced them in https://github.com/RevenueCat/purchases-android/pull/787/commits/66e86ee3e4b0dd6eb7a3c6906a84b954a2b2e48b and tests seem to pass fine 👍